subprojects/model-core/src/main/java/org/gradle/internal/extensibility/ConventionAwareHelper.java

```diff
@@ -60,14 +64,25 @@ private MappedProperty map(String propertyName, MappedPropertyImpl mapping) {
                 "You can't map a property that does not exist: propertyName=" + propertyName);
         }
 
-        if (_ineligiblePropertyNames.contains(propertyName)) {
+        // When there's a Property, use its `.convention()` method instead
```

Comment:
Do we need any "why" in the comment here or we will know that this is due to property upgrades?

Reply:
Good point, added some more explanation.